__tests__/login/oidc/TokenRequester.spec.ts

```diff
@@ -0,0 +1,244 @@
+    await TokenRefresher.request("global", {
+      grant_type: "refresh_token",
+      refresh_token: "thisIsARefreshToken"
+    });
+    /* eslint-enable @typescript-eslint/camelcase */
+    expect(defaultMocks.fetcher.fetch).toBeCalledWith(
+      IssuerConfigFetcherFetchConfigResponse.tokenEndpoint,
+      {
+        method: "POST",
+        headers: {
+          DPoP: DpopHeaderCreatorResponse,
+          "content-type": "application/x-www-form-urlencoded"
+        },
+        body:
+          "grant_type=refresh_token&refresh_token=thisIsARefreshToken&client_id=coolApp"
+      }
+    );
+  });
+
+  it("Adds an authorization header if a client secret is present", async () => {
+    setUpMockedReturnValues({
+      storageClientSecret: "theSecretSauceIsJustMayo"
+    });
+    const TokenRefresher = getTokenRequester();
+    /* eslint-disable @typescript-eslint/camelcase */
+    await TokenRefresher.request("global", {
+      grant_type: "refresh_token",
+      refresh_token: "thisIsARefreshToken"
+    });
+    /* eslint-enable @typescript-eslint/camelcase */
+    expect(defaultMocks.fetcher.fetch).toBeCalledWith(
+      IssuerConfigFetcherFetchConfigResponse.tokenEndpoint,
+      {
+        method: "POST",
+        headers: {
+          DPoP: DpopHeaderCreatorResponse,
+          Authorization: "Basic Y29vbEFwcDp0aGVTZWNyZXRTYXVjZUlzSnVzdE1heW8=",
+          "content-type": "application/x-www-form-urlencoded"
+        },
+        body:
+          "grant_type=refresh_token&refresh_token=thisIsARefreshToken&client_id=coolApp"
+      }
+    );
+  });
+
+  it("Fails elegantly if the idp returns a bad value", async () => {
+    setUpMockedReturnValues({
+      responseBody: JSON.stringify({
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        id_token: "ohNoThereIsNoAccessToken"
+      })
+    });
+    const TokenRefresher = getTokenRequester();
+    await expect(
+      /* eslint-disable @typescript-eslint/camelcase */
+      TokenRefresher.request("global", {
+        grant_type: "refresh_token",
+        refresh_token: "thisIsARefreshToken"
+      })
+      /* eslint-enable @typescript-eslint/camelcase */
+    ).rejects.toThrowError("IDP token route returned an invalid response.");
+  });
+
+  it("Fails elegantly if the issuer does not support refresh tokens", async () => {
+    setUpMockedReturnValues({
+      issuerConfig: {
+        ...IssuerConfigFetcherFetchConfigResponse,
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        grantTypesSupported: ["id_token"]
+      }
+    });
+    const TokenRefresher = getTokenRequester();
+    await expect(
+      /* eslint-disable @typescript-eslint/camelcase */
+      TokenRefresher.request("global", {
+        grant_type: "refresh_token",
+        refresh_token: "thisIsARefreshToken"
+      })
+      /* eslint-enable @typescript-eslint/camelcase */
+    ).rejects.toThrowError(
+      "The issuer https://idp.com does not support the refresh_token grant"
```

Comment:
Two points:
1) I always delimit placeholders in message strings with '[' and ']' (i.e. in this case the error should be `"The issuer [https://idp.com] does not ...`
2) I wouldn't hardcode the expected param value, I'd replace with the test value, which I think means: `The issuer [${defaultReturnValues.storageIdp}] does not...`